pkg/partmgr/partmgr.go

```diff
@@ -18,123 +18,203 @@ under the Apache 2.0 license is conditioned upon your compliance with
 such restriction.
 */
 
-// TODO: need to expand from a single partition to multiple
-
 package partmgr
 
 import (
+	"encoding/json"
 	"fmt"
+	"github.com/pkg/errors"
+	"github.com/v3io/v3io-go-http"
 	"github.com/v3io/v3io-tsdb/pkg/aggregate"
 	"github.com/v3io/v3io-tsdb/pkg/config"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"math"
+	"path"
+	"strconv"
 	"sync"
 )
 
-// Create new Partition Manager, for now confined to one Cyclic partition
-func NewPartitionMngr(cfg *config.DBPartConfig, path string) *PartitionManager {
-	newMngr := &PartitionManager{cfg: cfg, path: path, cyclic: true, ignoreWrap: true}
-	newMngr.headPartition = NewDBPartition(newMngr)
+// Create new Partition Manager
+func NewPartitionMngr(cfg *config.Schema, partPath string, cont *v3io.Container) *PartitionManager {
+	currentPartitionInterval, _ := utils.Str2duration(cfg.PartitionSchemaInfo.PartitionerInterval)
+	newMngr := &PartitionManager{cfg: cfg, path: partPath, cyclic: false, container: cont, currentPartitionInterval: currentPartitionInterval}
+	for _, part := range cfg.Partitions {
+		partPath := path.Join(newMngr.path, strconv.FormatInt(part.StartTime/1000, 10)) + "/"
```

Comment:
is this necessary to append "/" to the path?

Reply:
Yes, because Nginx expectes URLs to end with a slash. it would be either save the path here with a trailing slash or change every Nginx call and append the slash

Reply:
I prefer one place (close to HTTP request) that checks the URL and adds slash if missing

Reply:
that way we'll need to wrap the ngnix calls or alternatively do this before every call which is more error prone.